Index: post_phs/post_api.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>from request_phs.request_data import *\r\nfrom breakeven_price.monte_carlo import monte_carlo\r\n\r\ndef post_breakeven_price():\r\n    address = 'https://api.phs.vn/market/utilities.svc/PostBreakevenPrice'\r\n\r\n    tickers = ['AAA', 'ACB']\r\n    breakeven_price = dict()\r\n    for ticker in tickers:\r\n        try:\r\n            breakeven_price[ticker] = monte_carlo(ticker=ticker, graph='off')\r\n        except KeyError:\r\n            continue\r\n\r\n    json_str = json.dumps(breakeven_price)\r\n    r = requests.post(url=address, data=json_str,\r\n                      headers={'content-type': 'application/json'})\r\n\r\n    json.loads(r.json())\r\n\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/post_phs/post_api.py b/post_phs/post_api.py
--- a/post_phs/post_api.py	(revision b5a9505bac6556e0430456562749ad8cd90eefd2)
+++ b/post_phs/post_api.py	(date 1610690034442)
@@ -2,19 +2,32 @@
 from breakeven_price.monte_carlo import monte_carlo
 
 def post_breakeven_price():
-    address = 'https://api.phs.vn/market/utilities.svc/PostBreakevenPrice'
 
-    tickers = ['AAA', 'ACB']
-    breakeven_price = dict()
-    for ticker in tickers:
+    address = 'https://api.phs.vn/market/Utilities.svc/PostBreakevenPrice'
+    #tickers = request_ticker_list()
+    tickers = ['AAA', 'ACB', 'HBC', 'CTD', 'VCB']
+    breakeven_price = pd.Series(index=tickers, name='price', dtype=float)
+    for ticker in breakeven_price.index:
         try:
-            breakeven_price[ticker] = monte_carlo(ticker=ticker, graph='off')
+            price = monte_carlo(ticker=ticker, graph='off')
+            if price < 10000:
+                breakeven_price.loc[ticker] \
+                    = '{:,.0f}'.format(round(price,-1))
+            elif 10000 <= price < 50000:
+                breakeven_price.loc[ticker] \
+                    = '{:,.0f}'.format(50 * round(price/50))
+            else:
+                breakeven_price.loc[ticker] \
+                    = '{:,.0f}'.format(round(price,-2))
         except KeyError:
             continue
 
-    json_str = json.dumps(breakeven_price)
-    r = requests.post(url=address, data=json_str,
+    json_str = breakeven_price.to_json()
+
+    r = requests.post(url=address,
+                      data=json_str,
                       headers={'content-type': 'application/json'})
 
-    json.loads(r.json())
+    result = pd.DataFrame(json.loads(r.json()))
+
 
